src/eventsRelay.js

```diff
@@ -75,9 +75,9 @@ module.exports = class EventsRelay {
         );
         this.sendDelayed(tx);
       },
-      EpochLength: async event => {
+      EpochLength: async (event) => {
         const { epochLength } = event.returnValues;
-        const tx = Tx.epochLength(Number(epochLength));
+        const tx = Tx.epochLength(Number(epochLength), Number(event.blockNumber));
```

Comment:
We can save blockNumber in `bridgeState.epochLengths` as well and use it for tx validation 🤔

Reply:
I didn't get the idea. You propose to use these blockNumbers here when creating epochLength tx ?

Reply:
In bridgeState we [store all epochLength changes](https://github.com/leapdao/leap-node/blob/master/src/bridgeState.js#L134-L136). We can add blockNumber into this array (`[epochLength, blockNumber][]`) and use it in [this check](https://github.com/leapdao/leap-node/blob/fix/340-epoch-length/src/tx/applyTx/checkEpochLength.js)

Btw, do not forget to update check anyway — it should handle two tx types (now it handles zero :-))

Reply:
hm.. I've updated the check.

I don't see why we need to store blockNumber in bridgeState — we can just take it from the transaction (like I do in my code). If there is no blockNumber in epochLength, chance are that `checkEpochLength` won't be called at all, exactly the issue we are fixing here.

Reply:
or you mean, we can compare height from bridgeState with the one from tx ?

Reply:
Yes :-)